src/oidc.rs

```diff
@@ -0,0 +1,93 @@
+use crate::protobuf::envoy::config::cluster::v3::Cluster;
+use crate::protobuf::envoy::config::core::v3::http_uri::HttpUpstreamType;
+use crate::protobuf::envoy::config::core::v3::HttpUri;
+use crate::protobuf::envoy::config::route::v3::route_match::PathSpecifier;
+use crate::protobuf::envoy::config::route::v3::RouteMatch;
+use crate::protobuf::envoy::extensions::filters::http::jwt_authn::v3::jwt_provider::JwksSourceSpecifier;
+use crate::protobuf::envoy::extensions::filters::http::jwt_authn::v3::jwt_requirement::RequiresType;
+use crate::protobuf::envoy::extensions::filters::http::jwt_authn::v3::JwtAuthentication;
+use crate::protobuf::envoy::extensions::filters::http::jwt_authn::v3::JwtProvider;
+use crate::protobuf::envoy::extensions::filters::http::jwt_authn::v3::JwtRequirement;
+use crate::protobuf::envoy::extensions::filters::http::jwt_authn::v3::RemoteJwks;
+use crate::protobuf::envoy::extensions::filters::http::jwt_authn::v3::RequirementRule;
+use prost_types::Duration;
+
+use crate::envoy_helpers::get_envoy_cluster;
+use std::collections::HashMap;
+
+#[derive(Default)]
+pub struct OIDCConfig {
+    issuer: std::string::String,
+    audiences: Vec<std::string::String>,
+    certs: std::string::String,
+    cluster: std::string::String,
+}
+
+async fn example() -> i32 {
+    42
+}
+
+impl OIDCConfig {
+    pub fn new(issuer: std::string::String) -> OIDCConfig {
+        return OIDCConfig {
+            issuer: issuer,
+            audiences: Vec::new(),
+            ..Default::default()
```

Comment:
You can remove the `Vec::new` call too because `Default` is also implemented for `Vec` using an empty vector.